packages/edge-worker/src/EdgeWorker.ts

```diff
@@ -1022,45 +1014,120 @@ export class EdgeWorker extends EventEmitter {
 		// console.log('=== END WEBHOOK PAYLOAD ===')
 
-		await this.handleIssueUnassigned(webhook.notification.issue, repository);
+		await this.handleIssueUnassigned(issue, repository);
+	}
+
+	/**
+	 * Get cached repository for an issue (used by agentSessionPrompted Branch 3)
+	 */
+	private getCachedRepository(issueId: string): RepositoryConfig | null {
+		return this.repositoryRouter.getCachedRepository(
+			issueId,
+			this.repositories,
+		);
 	}
 
 	/**
-	 * Get Linear client for a workspace by finding first repository with that workspace ID
+	 * Get Linear client for a workspace
+	 *
+	 * Note: This is used by RepositoryRouter for project-based routing.
+	 * For platform-agnostic architecture, we return undefined for non-Linear platforms.
 	 */
 	private getLinearClientForWorkspace(
 		workspaceId: string,
 	): LinearClient | undefined {
-		for (const [repoId, repo] of this.repositories) {
+		// For Linear platform, get the client from LinearIssueTrackerService
+		for (const [_repoId, repo] of this.repositories) {
 			if (repo.linearWorkspaceId === workspaceId) {
-				return this.linearClients.get(repoId);
+				const issueTracker = this.issueTrackers.get(repo.id);
+				// Only LinearIssueTrackerService has a Linear client
+				if (issueTracker instanceof LinearIssueTrackerService) {
+					// Access private field via type assertion - needed for RepositoryRouter
+					return (issueTracker as any).linearClient;
+				}
 			}
 		}
 		return undefined;
 	}
 
+	/**
+	 * Post activity to Linear indicating which repository was selected and how
+	 */
+	private async postRepositorySelectionActivity(
+		agentSessionId: string,
+		repositoryId: string,
+		repositoryName: string,
+		routingMethod: string,
+	): Promise<void> {
+		const issueTracker = this.issueTrackers.get(repositoryId);
+		if (!issueTracker) {
+			console.warn(
+				`[EdgeWorker] Cannot post repository selection activity - no IssueTracker for repository ${repositoryId}`,
+			);
+			return;
+		}
+
+		const methodDescriptions: Record<string, string> = {
+			"label-based": "based on issue labels",
+			"project-based": "based on issue project",
+			"team-based": "based on issue team",
+			"team-prefix": "based on team identifier prefix",
+			"catch-all": "using catch-all configuration",
+			"workspace-fallback": "as workspace default",
+		};
+
+		const methodDescription =
+			methodDescriptions[routingMethod] || `using ${routingMethod}`;
+
+		try {
+			await issueTracker.createAgentActivity({
+				agentSessionId,
+				content: {
+					type: AgentActivityContentType.Thought,
+					body: `Selected repository: **${repositoryName}** (${methodDescription})`,
+				},
+			});
+			console.log(
+				`[EdgeWorker] Posted repository selection activity for ${repositoryName}`,
+			);
+		} catch (error) {
+			console.warn(
+				`[EdgeWorker] Failed to post repository selection activity:`,
+				error,
+			);
+			// Don't throw - this is a non-critical operation
+		}
+	}
+
```

Comment:
why is this added, there shouldnt be something like this added, compared to 'main'